test/e2e-secondary-network/framework.go

```diff
@@ -0,0 +1,77 @@
+// Copyright 2022 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package e2e
+
+import (
+	antreae2e "antrea.io/antrea/test/e2e"
+	corev1 "k8s.io/api/core/v1"
+	"log"
+	"os"
+	"path/filepath"
+	"time"
+)
+
+type TestData struct {
+	e2eTestData        *antreae2e.TestData
+	logsDirForTestCase string
+}
+
+func createDirectory(path string) error {
+	return os.Mkdir(path, 0700)
+}
+
+const (
+	busyboxImage    = "projects.registry.vmware.com/antrea/busybox"
+	defaultInterval = 1 * time.Second
+)
+
+var testData *TestData
+
+type ClusterInfo struct {
+	controlPlaneNodeName string
+}
+
+var clusterInfo ClusterInfo
+
+// podWaitForRunning polls the k8s apiserver until the specified Pod is in the "running" state (or
+// until the provided timeout expires).
+func (data *TestData) podWaitForRunning(timeout time.Duration, name, namespace string) error {
+	_, err := data.e2eTestData.PodWaitFor(timeout, name, namespace, func(pod *corev1.Pod) (bool, error) {
+		return pod.Status.Phase == corev1.PodRunning, nil
+	})
+	return err
+}
+
+func (data *TestData) setupLogDirectoryForTest(testName string) error {
```

Comment:
Why we need to re-implement this func, instead of reusing e2eTestData.setupLogDirectoryForTest()?

Reply:
setupLog**() was not exported at test/e2e. More over, we create few wrappers with references to how a similar scenario was handled at the multicluster test implementation.

Reply:
Maybe @antoninbas @luolanzone  know the background of the MC design.

I am fine to start with the current code anyway.

Reply:
I think a notable difference with the existing e2e tests is that the e2e-secondary-network tests assume that antrea has already been deployed, with the correct configuration (that's by design). So we cannot reuse `setupTest` for example.

That being said, I tend to agree with Jianjun about reusing `TestOptions` and `TestData` as is.

One thing that is also not clear to me is whether the e2e-secondary-network tests actually export any logs? In Antrea e2e tests, this is handled by https://github.com/antrea-io/antrea/blob/f7d70c35d6d8fdc4e614624254149d1cbe3a6a2f/test/e2e/fixtures.go#L277
But I don't see any code here that would export logs? I could be missing something.